maven-failsafe-plugin/src/main/java/org/apache/maven/plugin/failsafe/VerifyMojo.java

```diff
@@ -168,7 +167,10 @@ public class VerifyMojo extends AbstractMojo implements SurefireReportParameters
 
     private volatile PluginConsoleLogger consoleLogger;
 
-    @Inject
+    public VerifyMojo() {
+        this.logger = LoggerFactory.getLogger(getClass());
```

Comment:
In some place we initialize logger at field level and in other place in constructor.
For logger there is common to initialize it at field level.

I would like to have similar in all place.

Reply:
Agreed, normally I would do that here too, but to support tests, the constructor on line 174 accepts an externally instantiated Logger.  This constructor makes the Logger for non-test specified use.  I think it is preferred this way than to make the instance field non-final and have the other constructor discard the initial instance with the new one.  Please let me know if that's how you'd rather have it tho and I will change it.

Reply:
We did indeed used to use field injection. However, we've been moving to constructor injection as part of the move from Plexus to Guice, and in general (not just Maven) constructor injection is preferred for safety. Perhaps loggers are a special case, but I'm not sure.

Reply:
Yes, the Plexus logger was a component, so injecting was the way.  SLF4J is not, so agreed, Loggers this way are looked up with LoggerFactory and almost always on field initialization.

I found a couple of cases in these updates that tests set a mocked logger for verification, so kept that constructor and added a noargs constructor to instantiate it.  This balances the testing need and prod code need and minimizes instances/no temp objects.

Reply:
Now that you mention it, logging for verification makes me feel a bit icky. This feels like one mistake (injecting loggers) being used to duct tape over another mistake (unit tests that use loggers to inspect code). We should probably fix it at the root by not asserting on logger messages at all. However, that's a separate fix that shouldn't be included in this PR.

Reply:
This test uses the constructor: https://github.com/apache/maven-surefire/blob/master/maven-failsafe-plugin/src/test/java/org/apache/maven/plugin/failsafe/VerifyMojoTest.java#L45
Agreed it's outside this story's scope to consider updating.

(PluginConsoleLogger was the other "constructor with Logger arg" that I was vaguely remembering, not a test: https://github.com/apache/maven-surefire/blob/master/maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/log/PluginConsoleLogger.java#L38)

Reply:
ok, so leave where is needed for test. 
Agree we should not verify code by assertions on logging messages ... to fix in next steps.